src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -2,38 +2,53 @@ using Oceananigans.Advection:
     _advective_momentum_flux_Uu,
     _advective_momentum_flux_Uv,
     _advective_momentum_flux_Vu,
-    _advective_momentum_flux_Vv
+    _advective_momentum_flux_Vv,
+    advective_tracer_flux_x, 
+    advective_tracer_flux_y,
+    vertical_vorticity_U,
+    vertical_vorticity_V,
+    bernoulli_head_U,
+    bernoulli_head_V
 
 using Oceananigans.Grids: AbstractGrid
 using Oceananigans.Operators: Ax_qᶠᶜᶜ, Ay_qᶜᶠᶜ
 
+
 #####
 ##### Momentum flux operators
 #####
 
 @inline momentum_flux_huu(i, j, k, grid, advection, solution) =
-    @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
+    @inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, solution[1], solution[1]) / solution.h[i, j, k]
```

Comment:
What's your assessment about using `u` to advect `uh` rather than `uh` here?

Reply:
Is the idea that the following (which does not have the right syntax) would be more accurate?

```
@inbounds _advective_momentum_flux_Uu(i, j, k, grid, advection, u, solution[1])
```